tests/stackage_zlib_runpath/dynamic_libraries.bzl

```diff
@@ -0,0 +1,38 @@
+def _dynamic_libraries_impl(ctx):
+    outputs = []
+    solib_names = []
+    for target in ctx.attr.srcs:
+        cc_info = target[CcInfo]
+        for library_to_link in cc_info.linking_context.libraries_to_link.to_list():
+            library = library_to_link.resolved_symlink_dynamic_library
+            if not library or library.basename.find(ctx.attr.filter) == -1:
+                continue
+            outputs.append(library)
+            if library_to_link.dynamic_library:
+                solib_names.append(library_to_link.dynamic_library.short_path)
+    if ctx.attr.solib_names:
+        ctx.actions.write(
+            ctx.outputs.solib_names,
+            "\n".join(solib_names),
+        )
+    return [DefaultInfo(
+        files = depset(outputs),
+        runfiles = ctx.runfiles(files = outputs),
+    )]
+
+dynamic_libraries = rule(
```

Comment:
(When you reach enlightenment and whip up target rules just for a test. :ok_woman:)